src/fender_value.rs

```diff
@@ -272,10 +278,14 @@ impl FenderValue {
 
 /// Cast functions
 impl FenderValue {
-    pub fn cast_to(&self, typeid: FenderTypeId) -> FenderValue {
+    pub fn cast_to(
+        &self,
+        typeid: FenderTypeId,
+        engine: &mut ExecutionEngine<FenderTypeSystem>,
```

Comment:
I think this should be wrapped in an option so `cast_to` can be called on values when not in the context of the VM

Reply:
What happens when you try to cast to a list without the VM, then?

Reply:
Resolution: cast_to and cast_to_ctx, one which takes the ExecutionEngine and one which doesn't - cast_to_ctx will dispatch to cast_to in the default case